chain/vm/runtime.go

```diff
@@ -60,38 +61,62 @@ type Runtime struct {
 }
 
 func (rt *Runtime) TotalFilCircSupply() abi.TokenAmount {
-	total := types.FromFil(build.TotalFilecoin)
+
+	filVested, err := rt.vm.GetVestedFunds(rt.ctx)
+	if err != nil {
+		rt.Abortf(exitcode.ErrIllegalState, "failed to get vested funds for computing total supply: %s", err)
+	}
 
 	rew, err := rt.state.GetActor(builtin.RewardActorAddr)
 	if err != nil {
 		rt.Abortf(exitcode.ErrIllegalState, "failed to get reward actor for computing total supply: %s", err)
 	}
 
+	filMined := types.BigSub(types.FromFil(build.FilAllocStorageMining), rew.Balance)
+	if filMined.LessThan(big.Zero()) {
+		filMined = big.Zero()
+	}
```

Comment:
Consulting the reward actor's balance is not a great way of knowing how much it has produced, but until recently you couldn't do any better. Since https://github.com/filecoin-project/specs-actors/pull/809 you should just use the reward state's `TotalMined` field, which is a count of the amount actually produced.